### ISSUE TITLE
Switch xml2 to github mirror

### ISSUE THREAD

**Issue by petersjt014** [NixOS/nixpkgs#92909]
The tool [xml2](https://github.com/NixOS/nixpkgs/blob/84cf00f98031e93f389f1eb93c4a7374a33cc0a9/pkgs/tools/text/xml/xml2/default.nix) is currently fetched from, according to the package:

`url = "https://web.archive.org/web/20160427221603/http://download.ofb.net/gale/xml2-0.5.tar.gz"`

This does not feel like a great idea.

I suggest moving to the repo [here](https://github.com/clone/xml2), which seems to be identical anyway. This would both be good practice and make the tool itself updateable (I want to write it a man page eventually, once I reverse engineer it).

**Comment by samueldr** [comment]
Leaving some thoughts I left on IRC.

```
 <samueldr> I'm plenty comfortable with the source of xml2 being fetched from archive.org
 <samueldr> the hash is the same
 <samueldr> in fact, more comfortable than an arbitrary repo cloner
 <samueldr> now, what might be a better source with the optics of getting bug fixes is whatever debian is doing
 <samueldr> https://tracker.debian.org/pkg/xml2 -> https://salsa.debian.org/debian/xml2
 <samueldr> though other than an added manpage, it looks like the source is unchanged and unpatched
```

IMO, if archive.org becomes an issue, it could be changed to the debian.org repo. Or any other known trustable mirror for code. (Ideally using the exact same tarball as the release did)

**Comment by petersjt014** [comment]
Yeah, the [debian one](https://salsa.debian.org/debian/xml2/-/tree/master) does look better. Thanks for spotting that.
